devices/HumanStateProvider/HumanStateProvider.cpp

```diff
@@ -654,6 +670,221 @@ bool HumanStateProvider::open(yarp::os::Searchable& config)
 
     pImpl->baseTransformSolution.setRotation(iDynTree::Rotation::Identity());
 
+    // ================================================
+    // INITIALIZE CUSTOM CONSTRAINTS FOR INTEGRATION-IK
+    // ================================================
+    yarp::os::Bottle& constraintGroup = config.findGroup("CUSTOM_CONSTRAINTS");
+    if (constraintGroup.isNull()) {
+        yError() << LogPrefix << "Failed to find group CUSTOM_CONSTRAINTS";
+        return false;
+    }
```

Comment:
The `CUSTOM_CONSTRAINTS` may be set as an optional configuration (e.g. see optional configurations in https://github.com/robotology/human-dynamics-estimation/blob/devel/publishers/HumanStatePublisher/HumanStatePublisher.cpp#L110), so that we avoid to add dummy empty configurations in the `.xml` like in https://github.com/robotology/human-dynamics-estimation/pull/137/files#diff-3dd2ddd1d6acd982bda1dd8aa2d8c091R134-R149.

Reply:
Moreover, a warning or an error should be thrown in case this option is used, and the solver is not set to QP since those constraints will not be used.

Reply:
done!

Reply:
Probably we could also add a flag like use_custom_constraint. In this way it would be easier to pass from one solver to the other without commenting/uncommenting part of the code.

Reply:
Well, I am not sure about this option. Since the customized constraints group is used for QP method, and as you suggested it is optional now. So, if a user does not need this option, he/she can remove or comment it. For a specific robot/system, if we need it we will have the group almost always in xml file, otherwise we will remove the group  almost always for that robot! So, I do not see a real benefit for adding this option.